beetsplug/edit.py

```diff
@@ -262,10 +291,20 @@ def edit_objects(self, objs, fields):
                         return False
 
                 # Show the changes.
+                # If the objects are not on the DB yet, we need a copy of their
+                # original state for show_model_changes.
+                if all(not obj.id for obj in objs):
```

Comment:
Ah indeed, that does seem problematic. Maybe we should just be keeping a dictionary of data (i.e, `dict(obj)` instead of a full copy of the object?